tenseal/cpp/tensors/ckksvector.cpp

```diff
@@ -570,6 +570,60 @@ CKKSVector& CKKSVector::polyval_inplace(const vector<double>& coefficients) {
     return *this;
 }
 
+CKKSVector CKKSVector::conv2d_im2col(const vector<double>& kernel,
+                                     size_t windows_nb) {
+    CKKSVector new_vec = *this;
+    new_vec.conv2d_im2col_inplace(kernel, windows_nb);
+    return new_vec;
+}
+
+CKKSVector& CKKSVector::conv2d_im2col_inplace(const vector<double>& kernel,
+                                              size_t windows_nb) {
+    vector<double> plain_vec;
+    size_t chunck_size = windows_nb;
+    size_t chuncks_nb = kernel.size();
+
+    // check if vector size is not a power of 2
+    if (!(chuncks_nb && (!(chuncks_nb & (chuncks_nb - 1))))) {
+        throw invalid_argument("Kernel size should be a power of 2");
+    }
+
+    if (this->_size / windows_nb != chuncks_nb) {
+        throw invalid_argument("Matrix shape doesn't match with vector size");
+    }
+
+    plain_vec.reserve(this->_size);
+
+    for (size_t i = 0; i < chuncks_nb; i++) {
+        vector<double> tmp(chunck_size, kernel[i]);
+        plain_vec.insert(plain_vec.end(), tmp.begin(), tmp.end());
```

Comment:
```plain_vec.push_back``` might be more clear here.

Reply:
Actually, this will replicate each element in kernel vector for
e.g. if windows_nb == 4 and kernel == [1,2,3,4] => plain_vec == [1,1,1,1, 2,2,2,2, 3,3,3,3, 4,4,4,4]

I found out that this is the faster way to do that, is it correct?